### ISSUE TITLE
Children helper component has internal logic that makes it special

### ISSUE THREAD

**Issue by krambuhl** [krambuhl/rogain-resolve-tree#2]
Ideally components are not special. Components should all be treated equally.  

Currently the `<Children />` helper component requires some coordinated logic internally that makes it impossible/difficult to provide alternative composition components and transformations. 
## Plugin Based Resolution

A solution to this could be to add a transform layer for every component that passes through the resolution step.  this could allow a plugin to take a component tree and do deeper analysis and transformations than would be available to component helpers.
#### Slot Based Composition

MyLayout:

``` html
<div>
  <header><Slot name="header" /></header>
  <Slot name="contents" />
</div>
```

possible usage:

``` html
<MyLayout>
  <Slot name="header">Some Header</Slot>
  <Slot name="contents">
    <p>Lorem ipsum dolor sit amet, consectetur adipiscing elit. Maecenas maximus velit vel ligula tristique venenatis. Ut metus orci, posuere eu arcu ac, pellentesque tincidunt nisi.</p>
    <p>Lorem ipsum dolor sit amet, consectetur adipiscing elit. Maecenas maximus velit vel ligula tristique venenatis. Ut metus orci, posuere eu arcu ac, pellentesque tincidunt nisi.</p>
  </Slot>
</MyLayout>
```
#### Automate css best practices

A component (Button):

``` html
<a css-ns="button" href={@attrs.href}><Children /></a>
```

A typed element: (LargeDarkButton)

``` html
<Button css-mod="large color-dark"><Children /></Button>
```

Usage:

``` html
<LargeDarkButton href="#hello">Why Hello There.</LargeDarkButton>
```

Possible output:

``` html
<a class="button button--large button--color-dark" href="#hello">Why Hello There</a>
```

**Comment by krambuhl** [comment]
Decided to keep <Children /> internal. so slot based composition is not going to happen.

Transforms still could serve a purpose, but should be done at a tag level instead on component level.

**Comment by krambuhl** [comment]
The children/slot composition issue has been resolved, #4 addresses the attribute transforms.
